lib/selinux.c

```diff
@@ -98,6 +99,146 @@ int reset_selinux_file_context (void)
 	return 0;
 }
 
+/*
+ *  Log callback for libselinux internal error reporting.
+ */
+__attribute__((__format__ (printf, 2, 3)))
+static int selinux_log_cb (int type, const char *fmt, ...) {
+	va_list ap;
+	char *buf;
+	int r;
+#ifdef WITH_AUDIT
+	static int selinux_audit_fd = -2;
+#endif
+
+	va_start (ap, fmt);
+	r = vasprintf (&buf, fmt, ap);
+	va_end (ap);
+
+	if (r < 0) {
+		free (buf);
+		return 0;
+	}
+
+#ifdef WITH_AUDIT
+	if (-2 == selinux_audit_fd) {
+		selinux_audit_fd = audit_open ();
+	}
+
+	if (selinux_audit_fd < 0) {
+		/* You get these only when the kernel doesn't have
+		 * audit compiled in. */
+		if (   (errno != EINVAL)
+		    && (errno != EPROTONOSUPPORT)
+		    && (errno != EAFNOSUPPORT)) {
+
+		    (void) fputs (_("Cannot open audit interface.\n"),
+		              stderr);
+		    SYSLOG ((LOG_WARN, "Cannot open audit interface."));
+		}
+	} else {
+		uid_t audit_uid = geteuid ();
```

Comment:
Is this the right uid to use here?  Should we be using the real UID, or the login UID, or just pass 0 like systemd?

Reply:
Good question; From an SELinux standpoint the uid probably does not matter at all (only the context).

Reply:
Looks like dbus-broker uses getuid() i.e. real UID.  For shadow utils, that would be the UID of the caller even if the util is setuid-root, so that seems preferable to using geteuid().  Or you could ask what uid to use on linux-audit mailing list.

Reply:
switched to hardcoded `0`, like systemd, dbus and fedora's passwd